Odata-docs/webapi-8/fundamentals/client-driven-paging.md

```diff
@@ -0,0 +1,197 @@
+---
+title:  "Client-driven Paging in ASP.NET Core OData 8"
+description: Client-driven paging in ASP.NET Core OData 8.
+date:   2023-01-10
+ms.date: 1/10/2023
+author: habbes
+ms.author: clhabins
+---
+
+# Client-driven Paging in ASP.NET Core OData 8
+
+**Applies To**:[!INCLUDE[appliesto-webapi](../../includes/appliesto-webapi-v8.md)]
+
+A client can request the OData service to return a specific number of results. The `$top` query option is used to limit the number of results and the `skip` query option used to specify the offset. For example `GET /Customers?$skip=3$top=5` will return up to 5 items starting from the 4th item in the collection (i.e. after skipping the first 3).
+
+Using these two query options the client can divide a large collection into separate pages and fetch the pages in separate requests. For example, if our collection has 25 items, and we want each page to have 10 items, then we can fetch the different page as follows:
+
+- First page: `$top=10`
+- Second page: `$skip=10&$top=10`
+- Third page: `$skip=20&top=10` (The last page will return only 5 items)
+
+By default, ASP.NET Core OData 8 limits the maximum value for `$top` to 0. So if you add `$top=2` to your request, you would get an error with a message like:
+
+```json
+{
+    "error": {
+        "code": "",
+        "message": "The query specified in the URI is not valid. The limit of '0' for Top query has been exceeded. The value from the incoming request is '2'."
+}
+```
+
+## Configuring client-side paging in OData
+
+Configuring client-side paging involves two steps:
+
+1. Configuring the max `$top` query option value
+2. Adding `EnableQuery` attribute to the controller method
+
+You can configure the maximum top value using the [`ODataOptions.SetMaxTop()`](/dotnet/api/microsoft.aspnetcore.odata.odataoptions.setmaxtop) when adding OData services to your application. The following snippet would set the maximum allowed `$top` value to 100.
+
+```csharp
+services.AddOData(options =>
+    options.SetMaxTop(100)
+    //...
+    );
+```
+
+Now this will support values of up to 100, but return an error if `$top` exceeds 100.
+
+You can also remove the limit by setting it to `null`:
```

Comment:
does this mean when you set it to null you can use $top with whichever value and it will work.. I think you should say that explicitly if that's the case

Reply:
It means there's no (upper) limit. Any `$top` value >= 0 will be allowed.